sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -65,6 +67,10 @@ abstract class QueryGenerator(private val query: BindableQuery) {
         """.trimMargin(), argument.preparedStatementBinder(indexCalculator))
 
         precedingArrays.add(argument.name)
+
+        // Array arguments replace the argument in SQL, so minus 1 and then add the new number
+        // of arguments that are replacing it.
```

Comment:
Seems to be no longer true based on the code

Reply:
oh yea forgot to remove this comment. I just prune array args out of the initial count.